### PR TITLE
SMS non production

### PR DESCRIPTION
**Pull request by stephencookdev**
# Problem
Currently acceptors might get confused by just following the SMS sent to them when acceptance-testing (forgetting, or outright not knowing) that a current SMS limitation is that it will _always_ send the production URL, regardless of your actual environment

# Solution
Detect if you're running in a production environment, and just trigger a quick `alert` reminding you of the limitation, if attempting an SMS

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [X] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [n/a] Have new manual tests been written down?
- [X] Have tests passed locally?
- [n/a] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://569-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/crossDevice/CrossDeviceLink/index.js [hunk 1]
```diff
@@ -153,6 +153,13 @@ class CrossDeviceLinkUI extends Component {
   sendSms = () => {
     if (this.props.sms.valid) {
       this.setState({sending: true})
+
+      // add a quick note that this will send a production SMS, so non-production
+      // environment users will need to amend any URLs that they receive
+      if (!process.env.TRUE_PRODUCTION) {
```
Comment (stefaniacardenas) [inline src/components/crossDevice/CrossDeviceLink/index.js]: I'm scared that the name `TRUE_PRODUCTION` might create confusion and might start using it for the wrong reasons. Could we use the `development` environment for this? At the moment the `development` environment uses the same configuration as the test config but maybe they should be separate?

Comment (rfreitas) [reply src/components/crossDevice/CrossDeviceLink/index.js]: just to confirm `NODE_ENV == production` is equivalent to this, no?

Comment (stephencookdev) [reply src/components/crossDevice/CrossDeviceLink/index.js]: No, `NODE_ENV=production` is used in production and in staging.
We want something that targets everything _but_ production.

Comment (stephencookdev) [reply src/components/crossDevice/CrossDeviceLink/index.js]: So I've changed this to us having an `ENV` var that 1-to-1 maps with the name of the config block it comes from
i.e. `ENV` will equal either 'production', 'test', or 'staging'

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://569-pr-onfido-sdk-ui-onfido.surge.sh

File: webpack.config.babel.js [hunk 2]
```diff
@@ -102,13 +102,15 @@ const PROD_CONFIG = {
   'SMS_DELIVERY_URL': 'https://telephony.onfido.com',
   'PUBLIC_PATH' : `https://assets.onfido.com/web-sdk-releases/${packageJson.version}/`,
   'RESTRICTED_XDEVICE_FEATURE_ENABLED': true,
-  WOOPRA_DOMAIN
+  WOOPRA_DOMAIN,
+  'ENV': 'production'
```
Comment (rfreitas) [inline webpack.config.babel.js]: NODE_ENV is passed below, so this doesn't seem necessary (below as well)

Comment (stefaniacardenas) [reply webpack.config.babel.js]: Done

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://569-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/crossDevice/CrossDeviceLink/index.js [hunk 3]
```diff
@@ -153,6 +153,13 @@ class CrossDeviceLinkUI extends Component {
   sendSms = () => {
     if (this.props.sms.valid) {
       this.setState({sending: true})
+
+      // add a quick note that this will send a production SMS, so non-production
+      // environment users will need to amend any URLs that they receive
+      if (process.env.NODE_ENV !== 'production') {
+        alert(`An SMS will be sent, but the link in it will be to production, not to ${window.location.origin}`)
```
Comment (rfreitas) [inline src/components/crossDevice/CrossDeviceLink/index.js]: hmm, in reality nothing will be sent if it's in staging. so the text is not really accurate.

Comment (stefaniacardenas) [reply src/components/crossDevice/CrossDeviceLink/index.js]: I've added a condition for staging, with an alert informing devs that an SMS will not be sent and link must be copy-pasted.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-569-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://569-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**